tests/neo4j/txrun.py

```diff
@@ -40,3 +38,100 @@ def test_does_not_update_last_bookmark_on_rollback(self):
         tx.rollback()
         bookmarks = self._session.lastBookmarks()
         self.assertEqual(len(bookmarks), 0)
+
+    def test_does_not_update_last_bookmark_on_failure(self):
+        session = self._driver.session("w")
+        tx = session.beginTransaction()
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("RETURN").next()
+            tx.commit()
+        bookmarks = session.lastBookmarks()
+        self.assertEqual(len(bookmarks), 0)
+
+    def test_should_be_able_to_rollback_a_failure(self):
+        if get_driver_name() in ["go"]:
+            self.skipTest('Could not rollback transaction')
+        session = self._driver.session("w")
+        tx = session.beginTransaction()
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("RETURN").next()
+
+    def test_should_not_rollback_a_rollbacked_tx(self):
+        if get_driver_name() in ["go"]:
+            self.skipTest('Does not raise the exception')
+        self._session = self._driver.session(accessMode="w")
```

Comment:
```suggestion
        self._session = self._driver.session("w")
```
While possible, I prefer not to name positional arguments. This makes numerous advanced things like writing decorators that work with the arguments easier.